sql/logictest: TestLogic/local-declarative-schema/new_schema_changer/drop_database flake

Looks like there is some non-determinism in this test.

```
=== RUN   TestLogic/local-declarative-schema/new_schema_changer/drop_database
[14:54:47] --- progress: /go/src/github.com/cockroachdb/cockroach/pkg/sql/logictest/testdata/logic_test/inverted_index_multi_column: 55 statements
[14:54:47] --- done: /go/src/github.com/cockroachdb/cockroach/pkg/sql/logictest/testdata/logic_test/inverted_index_multi_column with config local-declarative-schema: 56 tests, 0 failures
[14:54:47] --- done: /go/src/github.com/cockroachdb/cockroach/pkg/sql/logictest/testdata/logic_test/inverted_filter_geospatial with config local-declarative-schema: 23 tests, 0 failures
=== CONT  TestLogic/local-declarative-schema/new_schema_changer/drop_database
    logic.go:2493: 
         
        /go/src/github.com/cockroachdb/cockroach/pkg/sql/logictest/testdata/logic_test/new_schema_changer:684: SELECT database_id, role_name, settings FROM system.database_role_settings ORDER BY 1, 2
        expected:
            database_id  role_name      settings
            0            ·              {application_name=d}
            0            test_set_role  {application_name=a,custom_option.setting=e}
            183          ·              {application_name=c}
            183          test_set_role  {application_name=b}
            
        but found (query options: "colnames") :
            database_id  role_name      settings
            0            ·              {application_name=d}
            0            test_set_role  {application_name=a,custom_option.setting=e}
            184          ·              {application_name=c}
            184          test_set_role  {application_name=b}
    logic.go:2036: 
         /go/src/github.com/cockroachdb/cockroach/pkg/sql/logictest/testdata/logic_test/new_schema_changer:693: too many errors encountered, skipping the rest of the input
[14:54:52] --- done: /go/src/github.com/cockroachdb/cockroach/pkg/sql/logictest/testdata/logic_test/new_schema_changer with config local-declarative-schema: 189 tests, 2 failures
            --- FAIL: TestLogic/local-declarative-schema/new_schema_changer/drop_database (5.20s)
```

--- a/pkg/sql/logictest/testdata/logic_test/new_schema_changer
+++ b/pkg/sql/logictest/testdata/logic_test/new_schema_changer
@@ -679,15 +679,19 @@ BLAH3
 BLAH4
 BLAH5
 
-# Verify that the current set of role settings
-query OTT colnames
-SELECT database_id, role_name, settings FROM system.database_role_settings ORDER BY 1, 2
+# Verify that the current set of role settings.
+# Note that the NULL name corresponds the ALL DATABASES configurations.
+query TTT colnames
+  SELECT name, role_name, settings
+    FROM system.database_role_settings
+         LEFT JOIN system.namespace AS ns ON database_id = ns.id
+ORDER BY name, role_name;
 ----
-database_id  role_name      settings
-0            ·              {application_name=d}
-0            test_set_role  {application_name=a,custom_option.setting=e}
-183          ·              {application_name=c}
-183          test_set_role  {application_name=b}
+name  role_name      settings
+NULL  ·              {application_name=d}
+NULL  test_set_role  {application_name=a,custom_option.setting=e}
+db1   ·              {application_name=c}
+db1   test_set_role  {application_name=b}
 
 statement error schema "sc1" is not empty and CASCADE was not specified
 DROP SCHEMA db1.sc1

